src/app/transformation/abstract_model.py

```diff
@@ -0,0 +1,76 @@
+import json
+from enum import Enum
+from typing import List
+
+
+ProgrammingStatement = str
+DependeeReference = str
+
+
+class Serializable:
+    def to_json(self):
+        return json.dumps(self, default=lambda o: o.__dict__, sort_keys=True, indent=4)
+
+
+class MetaInformation:
+    language: str
+    title: str
+    version: str
+
+    def __init__(self, language: str, title: str, version: str):
+        self.language = language
+        self.title = title
+        self.version = version
+
+
+class Dependee:
+    generated_reference: DependeeReference
+    initializer: ProgrammingStatement
+
+    def __init__(self, generated_reference: DependeeReference, initializer: ProgrammingStatement):
+        self.generated_reference = generated_reference
+        self.initializer = initializer
+
+
+class Setup:
+    independent: List[str]
+    dependees: List[Dependee]
+
+    def __init__(self, independent: List[str], dependees: List[Dependee]):
+        self.independent = independent
+        self.dependees = dependees
+
+
+class ParameterType(Enum):
+    STRING = 0
+    NUMBER = 1
+
+
+class NamedParameter:
+    abstract_name: str
+    generated_name: str
+    type: ParameterType
+    description: str
+
+    def __init__(self, abstract_name: str, generated_name: str, type: ParameterType, description: str):
+        self.abstract_name = abstract_name
+        self.generated_name = generated_name
+        self.type = type
+        self.description = description
+
+
+class Action:
+    abstract_name: str
+    generated_name: str
+    generated_actor: DependeeReference
```

Comment:
I know why you designed it like that, but I think we limit us a bit too much, when we only allow actions which have a predefined dependee. I'm thinking about calling static functions of a class from a library, e.g. `NiryoOne.Reset()` (completely made up). I would totally ignore that problem for now, but we should maybe investigate whether we need support for such calls.

I think we should allow actor to be empty in the long run to allow function calls without having an actor. But that would then also mean that the template needs to support it (see comment in the template. But I am also fine with tackling that later).